library/std/src/io/error/repr_bitpacked.rs

```diff
@@ -235,7 +235,13 @@ impl Repr {
 }
 
 impl Drop for Repr {
-    #[inline]
+    // This was #[inline] previously. Inlining the destructor of ErrorData is in
+    // no way helpful in real programs (as the source of the error will not be
+    // inlined, so there will not be any match assumptions to gain). The cost,
+    // meanwhile, is a code size increase by a factor of up to 5.4 in the case
+    // of dropping multiple io::Results in the same function
+    // (https://godbolt.org/z/8hfGchjsT).
+    #[inline(never)]
```

Comment:
Pondering: why is *this* the correct one to `inline(never)`?  What if we made `decode_repr` be `inline(never)` and allowed inlining the destructor as that turns into just a call to `decode_repr`?

Said otherwise, if this drop is a problem, wouldn't `data` and `data_mut` and such *also* not want this inlining?

Reply:
I do think it's worth double checking whether there is an option to get this behavior in more places the nontrivial drop may happen, as Scott mentioned. But if there doesn't seem to be a good way, I don't see any reason not to make this change given the perf results.

What is the result without any `#[inline]`/`#[inline(never)]` btw? Seems like that could give it some flexibility.